core/services/log/helpers_test.go

```diff
@@ -11,26 +25,215 @@ import (
+	helper.lb.SetLatestHeadFromStorage(nil)
+}
+
+func (helper *broadcasterHelper) register(listener log.Listener, contract log.AbigenContract, numConfirmations uint64) {
+	logs := []generated.AbigenLog{
+		flux_aggregator_wrapper.FluxAggregatorNewRound{},
+		flux_aggregator_wrapper.FluxAggregatorAnswerUpdated{},
+	}
+	helper.registerWithTopics(listener, contract, logs, numConfirmations)
+}
+
+func (helper *broadcasterHelper) registerWithTopics(listener log.Listener, contract log.AbigenContract, topics []generated.AbigenLog, numConfirmations uint64) {
+	unsubscribe := helper.lb.Register(listener, log.ListenerOpts{
+		Contract:         contract,
+		Logs:             topics,
+		NumConfirmations: numConfirmations,
+	})
+
+	helper.toUnsubscribe = append(helper.toUnsubscribe, unsubscribe)
+}
+
+func (helper *broadcasterHelper) unsubscribeAll() {
+	for _, unsubscribe := range helper.toUnsubscribe {
+		unsubscribe()
+	}
+	time.Sleep(100 * time.Millisecond)
+}
+func (helper *broadcasterHelper) stop() {
+	err := helper.lb.Stop()
+	require.NoError(helper.t, err)
+	helper.storeCleanup()
+}
+
+func newMockContract() *logmocks.AbigenContract {
+	addr := cltest.NewAddress()
+	contract := new(logmocks.AbigenContract)
+	contract.On("Address").Return(addr)
+	return contract
+}
+
+type logOnBlock struct {
+	logBlockNumber uint64
+	blockNumber    uint64
+	blockHash      common.Hash
+}
+
+func (l logOnBlock) String() string {
+	return fmt.Sprintf("blockInfo(log:%v received on: %v %s)", l.logBlockNumber, l.blockNumber, l.blockHash)
+}
+
+type received struct {
+	uniqueLogs []types.Log
+	logs       []types.Log
+	broadcasts []log.Broadcast
+	sync.Mutex
+}
+
+func newReceived(logs []types.Log) *received {
+	var rec received
+	rec.logs = logs
+	rec.uniqueLogs = logs
+	return &rec
+}
+
+func (rec *received) logsOnBlocks() []logOnBlock {
+	blocks := make([]logOnBlock, 0)
+	for _, broadcast := range rec.broadcasts {
+		blocks = append(blocks, logOnBlock{
+			logBlockNumber: broadcast.RawLog().BlockNumber,
+			blockNumber:    broadcast.LatestBlockNumber(),
+			blockHash:      broadcast.LatestBlockHash(),
+		})
+	}
+	return blocks
+}
+
 type simpleLogListener struct {
-	handler    func(lb log.Broadcast)
+	handler    func(lb log.Broadcast) bool
 	consumerID models.JobID
+	name       string
+	received   *received
 }
 
-func (listener simpleLogListener) HandleLog(lb log.Broadcast) {
-	listener.handler(lb)
+func newLogListener(t *testing.T, store *store.Store, name string) *simpleLogListener {
+	var rec received
+	return &simpleLogListener{
+		handler: func(lb log.Broadcast) bool {
+			return handleLogBroadcast(t, lb)
+		},
+		consumerID: createJob(t, store).ID,
+		name:       name,
+		received:   &rec,
+	}
+}
+
+func (listener *simpleLogListener) HandleLog(lb log.Broadcast) {
+	logger.Warnf(">>>>>>>>>>>>>>> Listener %v HandleLog for block %v %v received at %v %v", listener.name, lb.RawLog().BlockNumber, lb.RawLog().BlockHash, lb.LatestBlockNumber(), lb.LatestBlockHash())
```

Comment:
What's up with all the `>` s?

Reply:
Just for visibility in tests

Reply:
Ah I didn't see that this was a `_test.go` file